download/oci_dowload.go

```diff
@@ -0,0 +1,379 @@
+package download
+
+import (
+	"context"
+	"crypto/tls"
+	"encoding/base64"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/containerd/containerd/remotes/docker"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/pkg/content"
+	"oras.land/oras-go/pkg/oras"
+
+	"github.com/open-policy-agent/opa/bundle"
+	"github.com/open-policy-agent/opa/logging"
+	"github.com/open-policy-agent/opa/metrics"
+	"github.com/open-policy-agent/opa/plugins"
+	"github.com/open-policy-agent/opa/plugins/rest"
+	"github.com/open-policy-agent/opa/util"
+)
+
+type OCIDownloader struct {
+	config             Config                        // downloader configuration for tuning polling and other downloader behaviour
+	client             rest.Client                   // HTTP client to use for bundle downloading
+	path               string                        // path for OCI image as <registry>/<org>/<repo>:<tag>
+	localStorePath     string                        // path for the local OCI storage
+	trigger            chan chan struct{}            // channel to signal downloads when manual triggering is enabled
+	stop               chan chan struct{}            // used to signal plugin to stop running
+	f                  func(context.Context, Update) // callback function invoked when download updates occur
+	sizeLimitBytes     *int64                        // max bundle file size in bytes (passed to reader)
+	bvc                *bundle.VerificationConfig
+	respHdrTimeoutSec  int64
+	wg                 sync.WaitGroup
+	logger             logging.Logger
+	mtx                sync.Mutex
+	stopped            bool
+	persist            bool
+	longPollingEnabled bool
```

Comment:
How does long polling work with the OCI spec ?

Reply:
According to the [distribution specs](https://github.com/opencontainers/distribution-spec/blob/main/spec.md) at the moment there is no mention of pooling possibilities, but I left this in the OCI downloader as we might want to do some future improvements in the download section.

Reply:
Let's remove that code then. It still lives in the HTTP downloader.